features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,588 @@
+    const maxTokensToLiquidateNum = Number(maxTokensToLiquidate) || 0;
+
+    const sponsorPosition = activeSponsors[props.selectedSponsor];
+
+    const pendingWithdrawTimeRemaining =
+      Number(sponsorPosition.withdrawalTimestamp) - Number(currentTime);
+
+    const pendingTransferTimeRemaining =
+      Number(sponsorPosition.TransferTimestamp) - Number(currentTime);
+
+    const prettyTimeRemaining = (timeRemaining: number) => {
+      return timeRemaining > 0
+        ? Math.max(0, Math.floor(timeRemaining / 3600)) +
+            ":" +
+            Math.max(0, Math.floor((timeRemaining % 3600) / 60)) +
+            ":" +
+            Math.max(0, (timeRemaining % 3600) % 60)
+        : "None";
+    };
+    const pendingTransferTimeString = prettyTimeRemaining(
+      pendingTransferTimeRemaining
+    );
+    const pendingWithdrawTimeString = prettyTimeRemaining(
+      pendingWithdrawTimeRemaining
+    );
+
+    const underCollateralizedPrice =
+      Number(sponsorPosition.collateral) === 0 ||
+      Number(sponsorPosition.tokensOutstanding) === 0 ||
+      Number(collReq) === 0
+        ? 0
+        : Number(sponsorPosition.collateral) /
+          (Number(sponsorPosition.tokensOutstanding) *
+            parseFloat(fromWei(collReq)));
+
+    const underCollateralizedPercent =
+      Number(latestPrice) === 0
+        ? 0
+        : ((Number(latestPrice) - Number(underCollateralizedPrice)) /
+            Number(latestPrice)) *
+          100;
+
+    const collBalanceTooLow = () => {
+      if (tabIndex == "deposit") {
+        return collBalanceNum < collateralToDepositNum;
+      }
+      if (tabIndex == "liquidate") {
+        return collBalanceNum < finalFeeNum;
+      }
+    };
+
+    const tokenBalanceTooLow = tokenBalanceNum < maxTokensToLiquidateNum;
+
+    const needCollateralAllowance = () => {
+      if (tabIndex == "deposit") {
+        if (collAllowance === null || collateralToDeposit === null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < parseFloat(collateralToDeposit);
+      }
+      if (tabIndex == "liquidate") {
+        if (collAllowance === null || finalFee == null) return true;
+        if (collAllowance === "Infinity") return false;
+        return collAllowance < finalFee;
+      }
+      return true;
+    };
+    const needTokenAllowance = () => {
+      if (tokenAllowance === null || maxTokensToLiquidate === null) return true;
+      if (tokenAllowance === "Infinity") return false;
+      return tokenAllowance < parseFloat(maxTokensToLiquidate);
+    };
+
+    // Liquidation requires both collateral and synthetic to be approved. This
+    // results in a number of different configurations the user could be in depending
+    // on their interaction with this Dapp. Process the text here to keep the
+    // component simple.
+    const liquidationNeedAllowanceText = () => {
+      if (!needCollateralAllowance() && !needTokenAllowance()) {
+        return null;
```

Comment:
Since this function name implies that you're returning Text, why not return the empty string, "", as the default case?